src/dmd/expressionsem.d

```diff
@@ -1885,27 +1885,31 @@ private bool functionParameters(const ref Loc loc, Scope* sc,
              *   S().bar() = 1;      // bad!
              * }
              */
-            Dsymbol s = null;
-            if (fd.isThis() || fd.isNested())
-                s = fd.toParent2();
-            for (; s; s = s.toParent2())
+            bool checkEnclosingWild(Dsymbol s)
```

Comment:
Looks like both nested functions can be static ?

Reply:
if they didn't call `errorInout()`.